sentry_kafka_management/actions/local/kafka_cli.py

```diff
@@ -1 +1,189 @@
+    # Used when a dynamic config is set on a single broker
+    DYNAMIC_BROKER_CONFIG = "DYNAMIC_BROKER_CONFIG"
+    # Used when a dynamic config is set as a cluster-level default
+    DYNAMIC_DEFAULT_BROKER_CONFIG = "DYNAMIC_DEFAULT_BROKER_CONFIG"
+
+    @staticmethod
+    def values() -> set[str]:
+        """
+        Get all enum values.
+        """
+        return set(ConfigTypes._value2member_map_.keys())
+
+
+@dataclass
+class Config:
+    """
+    Represents a Config object parsed from the output of the
+    `kafka-configs` CLI.
+
+    Fields:
+        config_name: Name of the config
+        is_sensitive: Whether or not the config is sensitive
+                    (and therefore shouldn't have its value displayed)
+        active_value: The current active value of the config.
+                    Value hierarchy is: dynamic > static > default.
+        dynamic_value: Optional. The current dynamic value for the config (if one exists).
+        dynamic_default_value: Optional. The current cluster-wide dynamic value for the config
+                               (if one exists).
+        static_value: Optional. The current static value for the config (if one exists).
+        default_value: Optional. The current default value for the config (if one exists).
+    """
+
+    config_name: str
+    is_sensitive: bool
+    active_value: str
+    dynamic_value: str | None
+    dynamic_default_value: str | None
+    static_value: str | None
+    default_value: str | None
+
+
+def _str_to_bool(boolstr: str) -> bool:
+    """
+    Converts the string literal `"true"` or `"false"` to
+    its boolean value.
+    Raises an error if boolstr isn't a string representation of a bool.
+    """
+    if boolstr.lower() == "true":
+        return True
+    elif boolstr.lower() == "false":
+        return False
+    else:
+        raise ValueError(f"{boolstr} is not a string representation of a bool.")
+
+
+def _str_to_dict(dictstr: str) -> dict[str, str]:
+    """
+    Converts a string representation of a dict into a dict.
+    Differs from `json.loads()` by expecting the string to be in the format:
+    `"{key:config_name=value, key:config_name=value, ...}"`
+    """
+    # if dictstr is empty, return that
+    if dictstr == "{}":
+        return {}
+    if not dictstr.startswith("{") or not dictstr.endswith("}"):
+        raise ValueError(
+            'Expected string representation of dict in the format `"{key:value, key:value, ...}"`, '
+            f"instead got {dictstr}"
+        )
+    stripped = dictstr.strip("{}")
+
+    # regex that:
+    # - extracts the all-caps part of an entry as a key `([A-Z_]+)`
+    # - gets rid of the config name after the colon `:[^=]+=`
+    # - captures the value until we hit a comma, optional whitespace,
+    #   and all-caps chars `((?:(?!,\s*[A-Z_]+:).)+)`
+    regex = re.compile(r"([A-Z_]+):[^=]+=((?:(?!,\s*[A-Z_]+:).)+)")
+    res = {key: value.strip() for key, value in regex.findall(stripped)}
```

Comment:
I took a look at the output from kafka locally and the values didn't contain any delimiters, so this should work :+1: